### DISCUSSION TITLE
Help with @xstate/test using actors

### DISCUSSION THREAD

**Discussion by inverted-capital** [statelyai/xstate#4133] [Q&A]
Hello, does anyone have experience using @xstate/test to do model based testing using actors ?

I'm unsure if the library is intended to work with actors, but I cannot get the spawned actors to transition automatically during test paths - only the parent machine is walked in this way.

Is it possible to test like this, or have I misunderstood how this library works ?

I am modelling a system that tracks many simultaneous state machines, and so I figured the best way to test this was using actors to represent each "micro fsm" that the SUT represents internally.

**Comment by inverted-capital** [comment]
After some experimentation, it looks like actors are not ticked forwards when being used in the testing library.

**Reply by davidkpiano** [reply]
It doesn't work for actors yet, but we want to make that a valid use-case for `@xstate/test`.
